src/NHibernate/Type/TypeFactory.cs

```diff
@@ -522,6 +522,22 @@ public static IType HeuristicType(string typeName)
 			return HeuristicType(typeName, null);
 		}
 
+		/// <summary>
+		/// Uses heuristics to deduce a NHibernate type given a string naming the
+		/// type.
+		/// </summary>
+		/// <param name="type"></param>
+		/// <returns>An instance of <c>NHibernate.Type.IType</c></returns>
+		/// <remarks>
+		/// We check to see if it implements IType, ICompositeUserType, IUserType, ILifecycle (Association), or
+		/// IPersistentEnum.  If none of those are implemented then we will serialize the Type to the
+		/// database using NHibernate.Type.SerializableType(typeName)
+		/// </remarks>
+		public static IType HeuristicType(System.Type type)
+		{
+			return HeuristicType(type, parameters: null, length: null);
```

Comment:
Most of the logic in HeuristicType does not apply to `System.Type`

Reply:
What do you mean? I've tried to extract all type related logic in separate `HeuristicType` method. Do you see something unrelated in it?